cassandra/src/main/scala/org/apache/zeppelin/cassandra/CassandraInterpreterHelper.scala

```diff
@@ -0,0 +1,395 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zeppelin.cassandra
+
+import java.net.InetAddress
+import java.nio.ByteBuffer
+import java.util
+import java.util.Date
+
+import com.datastax.driver.core.DataType.Name._
+import com.datastax.driver.core._
+import com.datastax.driver.core.exceptions.DriverException
+import com.datastax.driver.core.policies.{LoggingRetryPolicy, FallthroughRetryPolicy, DowngradingConsistencyRetryPolicy, Policies}
+import org.apache.zeppelin.cassandra.TextBlochHierarchy._
+import org.apache.zeppelin.display.Input.ParamOption
+import org.apache.zeppelin.interpreter.InterpreterResult.Code
+import org.apache.zeppelin.interpreter.{InterpreterException, InterpreterResult, InterpreterContext}
+import org.slf4j.LoggerFactory
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+
+case class CassandraQueryOptions(consistency: Option[ConsistencyLevel],
+                                 serialConsistency:Option[ConsistencyLevel],
+                                 timestamp: Option[Long],
+                                 retryPolicy: Option[RetryPolicy],
+                                 fetchSize: Option[Int])
+
+object CassandraInterpreterHelper {
+  
+  val CHOICES_SEPARATOR : String = """\|"""
+  val VARIABLE_PATTERN = """\{\{[^}]+\}\}""".r
+  val SIMPLE_VARIABLE_DEFINITION_PATTERN = """\{\{([^=]+)=([^=]+)\}\}""".r
+  val MULTIPLE_CHOICES_VARIABLE_DEFINITION_PATTERN = """\{\{([^=]+)=((?:[^=]+\|)+[^|]+)\}\}""".r
+
+  val defaultRetryPolicy = Policies.defaultRetryPolicy()
+  val downgradingConsistencyRetryPolicy = DowngradingConsistencyRetryPolicy.INSTANCE
+  val fallThroughRetryPolicy = FallthroughRetryPolicy.INSTANCE
+  val loggingDefaultRetryPolicy = new LoggingRetryPolicy(defaultRetryPolicy)
+  val loggingDownGradingRetryPolicy = new LoggingRetryPolicy(downgradingConsistencyRetryPolicy)
+  val loggingFallThrougRetryPolicy = new LoggingRetryPolicy(fallThroughRetryPolicy)
+
+  val preparedStatements : mutable.Map[String,PreparedStatement] = mutable.Map[String,PreparedStatement]()
```

Comment:
it could be worth moving this to the instance. The object is Classloader global, so are its vals.
Using a Synchronized map  instead of the default HashMap implem will avoid race conditions or looking into scala.collection.concurrent.TrieMap could be interesting